source/common/ur_util.h

```diff
@@ -76,4 +80,54 @@ inline bool getenv_tobool( const char* name )
     return 0 == strcmp( "1", env );
 }
 
+inline int parse_env_var(const char *env_var_name,
+                         std::unordered_map<std::string, std::vector<std::string>> &map) {
+    // example: ENV_VAR="param_1:value_1,value_2;param_2:value_1"
+    char main_delim = ';';
+    char key_value_delim = ':';
+    char values_delim = ',';
+    int delim_pos = 0;
+
+    const char *env_var_c_str = std::getenv(env_var_name);
+    if (env_var_c_str == nullptr) {
+        std::cout << "Environment variable " << env_var_name << " is not set!" << std::endl;
+        return 0;
+    }
+    std::string env_var(env_var_c_str);
+
+    // check if the env var is in the simple format
+    // example: ENV_VAR=1
+    if (env_var.find(':') == std::string::npos) {
+        std::vector<std::string> value_vec;
+        value_vec.push_back(env_var);
+        map[env_var_name] = value_vec;
+        return 0;
+    }
+
+    std::stringstream ss(env_var);
+    std::string key_value;
+    while (std::getline(ss, key_value, main_delim)) {
+        std::string key;
+        std::string values;
+        std::stringstream kv_ss(key_value);
+        std::getline(kv_ss, key, key_value_delim);
+        std::getline(kv_ss, values);
+        if (values.empty()) {
+            std::cout << "Wrong format of the " << env_var_name << " environment variable! Value not found!" << std::endl;
+            std::cout << "Proper format is: ENV_VAR=\"param_1:value_1,value_2;param_2:value_1\"" << std::endl;
+            return -1;
```

Comment:
You can consider just throwing an exception in this case. Then, you don't need to return any status and you could just return the map by value, instead of using a reference.

Reply:
std::optional requires C++17